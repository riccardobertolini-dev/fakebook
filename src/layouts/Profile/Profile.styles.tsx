import styled, { css } from 'styled-components';
import { Link } from 'react-router-dom';

export const ProfileNavBar = styled.div`
	width: 100%;
	padding: 20px 0;
	display: flex;
	justify-content: center;

	a,
	a:visited {
		text-decoration: none;
		color: #fff;
		background-color: #4267b2;
		display: block;
		padding: 5px 20px;
		border-radius: 25px;
		border: 1px solid #4267b2;
	}
`;

export const ProfileWrapper = styled.div`
	background-color: #fff;
	width: 100%;
	padding: 80px 0;
	border: 1px solid grey;
	border-radius: 25px;
	box-shadow: 2px 2px 10px #eee;
	position: relative;
	overflow: hidden;
`;

export const ProfilePic = styled.img`
	border-radius: 5px;
	position: relative;
	z-index: 1;
`;

export const ProfileName = styled.div`
	margin-top: 20px;
	font-weight: 800;
	font-size: 30px;
	font-family: 'Lato', sans-serif;
	position: relative;
	z-index: 1;
`;

export const ProfileHeader = styled.div`
	display: flex;
	justify-content: center;
	flex-direction: column;
	align-items: center;
	padding-top: 60px;
`;

export const infoDesign = css`
	margin-top: 20px;
	font-weight: 200;
	font-size: 15px;
	font-family: 'Lato', sans-serif;
`;

export const ProfileAddress = styled.div`
	border-top: 1px solid #ccc;
	padding-top: 20px;
	${infoDesign};
`;

export const ProfileDetails = styled.div`
	${infoDesign};
	position: relative;
	z-index: 1;
`;

export const BackgroundImage = styled.div<{ src: string }>`
	background-image: url(${props => props.src});
	height: 250px;
	width: 100%;
	position: absolute;
	top: 0;
	z-index: 0;
`;

export const ClosingButton = styled(Link)`
	position: absolute;
	top: 20px;
	right: 20px;
	font-size: 30px;
	z-index: 10;
	color: white;
`;
