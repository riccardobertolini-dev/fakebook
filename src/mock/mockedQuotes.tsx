export const MockedQuote1 = {
	text: 'Genius is one percent inspiration and ninety-nine percent perspiration.',
	author: 'Thomas Edison',
};
export const MockedQuote2 = {
	text: 'You can observe a lot just by watching.',
	author: 'Yogi Berra',
};
