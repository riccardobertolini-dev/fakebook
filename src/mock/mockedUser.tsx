export const MockedUser1 = {
	gender: 'female',
	name: {
		title: 'Ms',
		first: 'Isabelle',
		last: 'Davies',
	},
	location: {
		street: {
			number: 8255,
			name: 'Esk Street',
		},
		city: 'Taupo',
		state: 'West Coast',
		country: 'New Zealand',
		postcode: 25207,
		coordinates: {
			latitude: '53.3834',
			longitude: '16.3516',
		},
		timezone: {
			offset: '-6:00',
			description: 'Central Time (US & Canada), Mexico City',
		},
	},
	email: 'isabelle.davies@example.com',
	login: {
		uuid: '71d3d4bf-2c47-450f-b3bf-e15a7fef23da',
		username: 'yellowtiger203',
		password: 'pyramid',
		salt: 'WrEua7np',
		md5: '8d185b252bdc6739cc36a7d4c67bcb1f',
		sha1: '4843e0607dd75b952504e278013508847f002237',
		sha256: 'dde6119b6dad7740579c5ace7f89789b7c24973dd203285bac2028018a046926',
	},
	dob: {
		date: '1965-10-08T18:06:10.878Z',
		age: 57,
	},
	registered: {
		date: '2015-12-03T15:58:10.458Z',
		age: 6,
	},
	phone: '(999)-128-6616',
	cell: '(132)-372-7000',
	id: {
		name: '',
		value: null,
	},
	picture: {
		large: 'https://randomuser.me/api/portraits/women/44.jpg',
		medium: 'https://randomuser.me/api/portraits/med/women/44.jpg',
		thumbnail: 'https://randomuser.me/api/portraits/thumb/women/44.jpg',
	},
	nat: 'NZ',
};
