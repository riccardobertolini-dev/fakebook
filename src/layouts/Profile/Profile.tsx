import { Link, useParams, Navigate } from 'react-router-dom';
import { useGlobalContext } from '../../globalContext';
import { ProfilePic, ProfileWrapper, ProfileNavBar, ProfileHeader, ProfileName, ProfileDetails, ProfileAddress } from './Profile.styles';
import { AiOutlineHome, AiOutlineMail, AiOutlinePhone } from 'react-icons/ai';

const Profile = () => {
	const { userSlug } = useParams();
	const { users } = useGlobalContext();
	const user = users.filter(user => user.login.uuid === userSlug)[0] || null;

	if (!user) return <Navigate to="/" />;

	return (
		<>
			<ProfileWrapper>
				<ProfileHeader>
					<ProfilePic src={user.picture.large} alt="" />
					<ProfileName>
						{user.name.title} {user.name.first} {user.name.last}
					</ProfileName>
					<ProfileAddress>
						<AiOutlineHome />
						<span> </span>
						{user.location.street.name} {user.location.street.number}, {user.location.city} ({user.location.country})
					</ProfileAddress>
					<ProfileDetails>
						{' '}
						<AiOutlineMail /> <Link to={`mailto:${user.email}`}>{user.email}</Link>
					</ProfileDetails>
					<ProfileDetails>
						<AiOutlinePhone /> {user.phone}
					</ProfileDetails>
				</ProfileHeader>
				<br />
				<ProfileNavBar>
					<Link to={'/'}>Back to Home</Link>
				</ProfileNavBar>
			</ProfileWrapper>
		</>
	);
};

export default Profile;
