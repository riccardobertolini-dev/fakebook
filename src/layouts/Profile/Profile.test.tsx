import renderer from 'react-test-renderer';
import React from 'react';
import Profile from './Profile';
import { MockedUser1 } from '../../mock/mockedUser';

let mockedUsedNavigate: () => void;

jest.mock('react-router-dom', () => {
	return {
		...jest.requireActual('react-router-dom'),
		Link: jest.fn(),
		useParams: () => ({
			userSlug: '71d3d4bf-2c47-450f-b3bf-e15a7fef23da',
		}),
		Navigate: jest.fn().mockReturnValueOnce(() => {
			mockedUsedNavigate = jest.fn();
			return mockedUsedNavigate;
		}),
	};
});

jest.mock('../../globalContext', () => ({
	useGlobalContext: () => ({
		users: [MockedUser1],
	}),
}));

describe('Profile component', () => {
	it('should render the user profile component', () => {
		const wrapper = renderer.create(<Profile />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
