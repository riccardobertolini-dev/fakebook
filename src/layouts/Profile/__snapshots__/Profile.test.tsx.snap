// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should render the user profile component 1`] = `
<div
  className="sc-gswNZR iXHHNU"
>
  <div
    className="sc-eDvSVe cKXiIf"
  >
    <img
      alt=""
      className="sc-dkrFOg tNBPJ"
      src="https://randomuser.me/api/portraits/women/44.jpg"
    />
    <div
      className="sc-hLBbgP hCowxi"
    >
      Ms
       
      Isabelle
       
      Davies
    </div>
    <div
      className="sc-jSUZER jaWJmJ"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
        />
      </svg>
      <span>
         
      </span>
      Esk Street
       
      8255
      , 
      Taupo
       (
      New Zealand
      )
    </div>
    <div
      className="sc-gKPRtg hgwDUe"
    >
       
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
        />
      </svg>
       
    </div>
    <div
      className="sc-gKPRtg hgwDUe"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M877.1 238.7L770.6 132.3c-13-13-30.4-20.3-48.8-20.3s-35.8 7.2-48.8 20.3L558.3 246.8c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l89.6 89.7a405.46 405.46 0 0 1-86.4 127.3c-36.7 36.9-79.6 66-127.2 86.6l-89.6-89.7c-13-13-30.4-20.3-48.8-20.3a68.2 68.2 0 0 0-48.8 20.3L132.3 673c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l106.4 106.4c22.2 22.2 52.8 34.9 84.2 34.9 6.5 0 12.8-.5 19.2-1.6 132.4-21.8 263.8-92.3 369.9-198.3C818 606 888.4 474.6 910.4 342.1c6.3-37.6-6.3-76.3-33.3-103.4zm-37.6 91.5c-19.5 117.9-82.9 235.5-178.4 331s-213 158.9-330.9 178.4c-14.8 2.5-30-2.5-40.8-13.2L184.9 721.9 295.7 611l119.8 120 .9.9 21.6-8a481.29 481.29 0 0 0 285.7-285.8l8-21.6-120.8-120.7 110.8-110.9 104.5 104.5c10.8 10.8 15.8 26 13.3 40.8z"
        />
      </svg>
       
      (999)-128-6616
    </div>
  </div>
  <br />
  <div
    className="sc-bcXHqe ccRnkD"
  />
</div>
`;
